library/src/scala/annotation/publicInBinary.scala

```diff
@@ -0,0 +1,14 @@
+package scala.annotation
+
+/** A binary API is a definition that is annotated with `@publicInBinary` or overrides a definition annotated with `@publicInBinary`.
+ *  This annotation can be placed on `def`, `val`, `lazy val`, `var`, `object`, and `given` definitions.
+ *  A binary API will be publicly available in the bytecode.
```

Comment:
Can we clarify the behavior of this annotation in this documentation? As far as I can tell, we've established that this annotation doesn't change the bytecode representation of the thing that is annotated, because everything that is allowed to be annotated will already be public, so the documentation as-is might be confusing. Instead the effect of this annotation is:
- no unnecessary accessors will be generated/used (which I guess should be considered a bug-fix since these accessors were never actually necessary and could break compilation?)
- Tools like mima/tasty-mima can rely on this annotation to determine if it is safe for Scala binary-compatibility to change/remove a method that is public in the bytecode.

Reply:
Rewrote part of the doc